migration-0.55-data/main.go

```diff
@@ -0,0 +1,399 @@
+			return fmt.Errorf("StatementAppendLogsV2:%w", err)
+		}
+
+	}
+
+	insertResourcesStmtV2, err = destConn.PrepareBatch(
+		ctx,
+		fmt.Sprintf("INSERT into %s", destResourceTableName),
+		driver.WithReleaseConnection(),
+	)
+	if err != nil {
+		return fmt.Errorf("couldn't PrepareBatch for inserting resource fingerprints :%w", err)
+	}
+
+	for bucketTs, resources := range resourcesSeen {
+		for resourceLabels, fingerprint := range resources {
+			insertResourcesStmtV2.Append(
+				resourceLabels,
+				fingerprint,
+				bucketTs,
+			)
+		}
+	}
+
+	var wg sync.WaitGroup
+	chErr := make(chan error, 2)
+	chDuration := make(chan statementSendDuration, 2)
+
+	wg.Add(2)
+
+	go send(insertLogsStmtV2, destTableName, chDuration, chErr, &wg)
+	go send(insertResourcesStmtV2, destResourceTableName, chDuration, chErr, &wg)
+
+	wg.Wait()
+	close(chErr)
+
+	// check the errors
+	for i := 0; i < 2; i++ {
+		if r := <-chErr; r != nil {
+			return fmt.Errorf("StatementSend:%w", r)
+		}
+	}
+
+	for i := 0; i < 2; i++ {
+		sendDuration := <-chDuration
+		zap.L().Info("Send duration", zap.String("table", sendDuration.Name), zap.Duration("duration", sendDuration.duration))
+	}
+
+	return nil
+}
+
+func send(statement driver.Batch, tableName string, durationCh chan<- statementSendDuration, chErr chan<- error, wg *sync.WaitGroup) {
+	defer wg.Done()
+	start := time.Now()
+	err := statement.Send()
+	chErr <- err
+	durationCh <- statementSendDuration{
+		Name:     tableName,
+		duration: time.Since(start),
+	}
+}
+
+func main() {
+	logger, _ := zap.NewDevelopment()
+	defer logger.Sync()
+	undo := zap.ReplaceGlobals(logger)
+	defer undo()
+
+	start := time.Now()
+	hostFlag := flag.String("host", "127.0.0.1", "clickhouse host")
+	portFlag := flag.String("port", "9000", "clickhouse port")
+	userNameFlag := flag.String("userName", "default", "clickhouse username")
+	passwordFlag := flag.String("password", "", "clickhouse password")
+	batchDuration := flag.Int("batch_duration", 5, "batch duration in minutes")
+	batchSize := flag.Int("batch_size", 30000, "clickhouse password")
+	startTimestamp := flag.Int64("start_ts", 0, "start timestamp in ns")
+	endTimestamp := flag.Int64("end_ts", 0, "end timestamp in ns")
+	sourceDbName := flag.String("source_db", "signoz_logs", "database name")
+	destDbName := flag.String("dest_db", "signoz_logs", "dest database name")
+	sourceTableName := flag.String("source_table", "distributed_logs", "table name")
+	destTableName := flag.String("dest_table", "distributed_logs_v2", "dest table name")
+	destResourceTableName := flag.String("resource_table", "distributed_logs_v2_resource", "dest resource table name")
+
+	flag.Parse()
+
+	sourceConn, err := connect(*hostFlag, *portFlag, *userNameFlag, *passwordFlag, *sourceDbName)
+	if err != nil {
+		zap.S().Fatal("Error while connecting to clickhouse", zap.Error(err))
+	}
+	defer sourceConn.Close()
+
+	destConn, err := connect("localhost", *portFlag, *userNameFlag, *passwordFlag, *destDbName)
```

Comment:
```suggestion
	destConn, err := connect(*hostFlag, *portFlag, *userNameFlag, *passwordFlag, *destDbName)
```
?

Reply:
Ahh my bad,  updated the flags for dest connection where user might use a different clickhouse cluster.